src/eval.cc

```diff
@@ -332,7 +332,9 @@ void Evaluator::EvalAssign(const AssignStmt* stmt) {
   Var* var =
       EvalRHS(lhs, stmt->rhs, stmt->orig_rhs, stmt->op,
               stmt->directive == AssignDirective::OVERRIDE, &needs_assign);
-  if (needs_assign) {
+  // Ignore any attempts to set .SHELLSTATUS, it's a built-in variable
```

Comment:
nit: separate the check for `.SHELLSTATUS` into a separate if statement because it is very different thing from checking `needs_assign`. 

I suppose you don't need to call `var-SetReadonly()` in either case, so this could look like this:

```
if (!needs_assign) {
  return;
}

if (lhs.str()  == ".SHELLSTATUS") {
  return;
}

bool readonly;
...
```